config.go

```diff
@@ -162,14 +162,14 @@ func (config TGFConfig) String() string {
 
 var cachedAwsConfig *aws.Config
 
-func (tgfConfig *TGFConfig) getAwsConfig(assumeRoleDuration time.Duration) (*aws.Config, error) {
+func (tgfConfig *TGFConfig) getAwsConfig(assumeRoleDuration time.Duration) (aws.Config, error) {
```

Comment:
The downside of not using a pointer here is we lose the ability to return nil alongside an error. We are now returning a possibly pretty broken config alongside an error. The caller still has to error check in either cases, but we now return some form of config anyway. I guess the source of this change was the existence of config and _config variables handling the pointer and content of it handled separately, there's probably a way to achieve both upsides here. I know Boris was taking a look at this as well.

Reply:
The problem is that AWS changed the way they are dealing with that kind of objects. Previously, they were pointers and they aren't anymore. That's not a problem to return the broken structure alongside with the error IMO.

Reply:
:+1: